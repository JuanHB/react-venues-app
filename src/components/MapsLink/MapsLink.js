import React from 'react';
import PropTypes from 'prop-types';
import { createMapsUrl } from 'src/utils/createMapsUrl'

const MapsLink = ({ address }) => {

  return (
    <a
      target='_blank'
      className='card-link'
      rel='noopener noreferrer'
      href={createMapsUrl(address)}>
      How to get there
    </a>
  );

};

MapsLink.propTypes = {
  address: PropTypes.string.isRequired
};

export default MapsLink;