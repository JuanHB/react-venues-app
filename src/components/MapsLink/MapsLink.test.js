import React from 'react';
import renderer from 'react-test-renderer';
import MapsLink from './MapsLink';

const address = 'New York, NY';

it('renders correctly MapsLink component', () => {
  const MapsLinkComponent = renderer.create(
    <MapsLink address={address} />
  ).toJSON();
  expect(MapsLinkComponent).toMatchSnapshot();
});