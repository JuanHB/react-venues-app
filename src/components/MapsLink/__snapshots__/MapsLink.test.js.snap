// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly MapsLink component 1`] = `
<a
  className="card-link"
  href="https://www.google.com/maps/search/?api=1&query=New York, NY"
  rel="noopener noreferrer"
  target="_blank"
>
  How to get there
</a>
`;
